Adding decoding async value for `<Image />` 🖼 (#1969)
Shopify/hydrogen: docs/components/primitive/image.md

| height         | <code>height &#124; string</code>                              | The integer or string value for the height of the image. This is a required prop when `src` is present.                                                                                                        |
| loader?        | <code>(props: ShopifyLoaderParams &#124; LoaderOptions) => string</code> | A custom function that generates the image URL. Parameters passed in are either `ShopifyLoaderParams` if using the `data` prop, or the `LoaderOptions` object that you pass to `loaderOptions`. |
| loaderOptions? | <code>ShopifyLoaderOptions &#124; LoaderOptions</code>       | An object of `loader` function options. For example, if the `loader` function requires a `scale` option, then the value can be a property of the `loaderOptions` object (for example, `{scale: 2}`). When the `data` prop is used, the object shape will be `ShopifyLoaderOptions`. When the `src` prop is used, the data shape is whatever you define it to be, and this shape will be passed to `loader`. |

> | widths?         | <code>(number &#124; string)[]</code>                              | An array of pixel widths to overwrite the default generated srcset. For example, `[300, 600, 800]`. It only applies to images from Shopify CDN. 

Reviewer: ```suggestion
| widths?         | <code>(number &#124; string)[]</code>                              | An array of pixel widths to overwrite the default-generated `srcset`. For example, `[300, 600, 800]`. This prop only applies to images from Shopify CDN. 
```

> | decoding?         | <code>('async' &#124; 'sync' &#124; 'auto')</code>                              | The decoding property of the HTMLImageElement interface represents a hint given to the browser on how it should decode the image. Defaults to `async` .    

Reviewer: ```suggestion
| decoding?         | <code>('async' &#124; 'sync' &#124; 'auto')</code>                              | The decoding property of the `HTMLImageElement` interface represents a hint given to the browser on how to decode the image. Defaults to `async` .    
```
